client/json-rpc/rpc_procedures.c

```diff
@@ -0,0 +1,328 @@
+ "version":"1.1", "method":"do_xfer", "params":["S","A","sign"], "id":1
+ 
+ reponse:
+ "result":[{"block":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"}], "error":null, "id":1
+ "jsonrpc":"2.0", "result":[{"block":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"}], "error":null, "id":1
+ "version":"1.1", "result":[{"block":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"}], "error":null, "id":1
+ */
+
+cJSON * method_xdag_do_xfer(struct xdag_rpc_context * ctx, cJSON * params, cJSON *id, char *version);
+cJSON * method_xdag_do_xfer(struct xdag_rpc_context * ctx, cJSON * params, cJSON *id, char *version)
+{
+	//todo: need password or not?
+	xdag_debug("rpc call method do_xfer, version %s", version);
+	
+	char amount[128] = {0};
+	char address[128] = {0};
+	
+	if (params) {
+		if (cJSON_IsArray(params) && cJSON_GetArraySize(params) == 1) {
+			cJSON* param = cJSON_GetArrayItem(params, 0);
+			if (!param || !cJSON_IsObject(param)) {
+				ctx->error_code = 1;
+				ctx->error_message = strdup("Invalid parameters.");
+				return NULL;
+			}
+			
+			cJSON * json_amount = cJSON_GetObjectItem(param, "amount");
+			if (cJSON_IsString(json_amount)) {
+				strcpy(amount, json_amount->valuestring);
+			}
+			
+			cJSON * json_address = cJSON_GetObjectItem(param, "address");
+			if (cJSON_IsString(json_address)) {
+				strcpy(address, json_address->valuestring);
+			}
+		} else {
+			ctx->error_code = 1;
+			ctx->error_message = strdup("Invalid parameters.");
+			return NULL;
+		}
+	} else {
+		ctx->error_code = 1;
+		ctx->error_message = strdup("Invalid parameters.");
+		return NULL;
+	}
+	
+	if(g_xdag_state < XDAG_STATE_XFER) {
+		ctx->error_code = 1;
+		ctx->error_message = strdup("Not ready to transfer.");
+		return NULL;
+	} else {					
+		struct xfer_callback_data xfer;
+		
+		memset(&xfer, 0, sizeof(xfer));
+		xfer.remains = xdags2amount(amount);
+		if(!xfer.remains) {
+			ctx->error_code = 1;
+			ctx->error_message = strdup("Xfer: nothing to transfer.");
+		} else if(xfer.remains > xdag_get_balance(0)) {
+			ctx->error_code = 1;
+			ctx->error_message = strdup("Xfer: balance too small.");
+		} else if(xdag_address2hash(address, xfer.fields[XFER_MAX_IN].hash)) {
+			ctx->error_code = 1;
+			ctx->error_message = strdup("Xfer: incorrect address.");
+		} else {
+			xdag_wallet_default_key(&xfer.keys[XFER_MAX_IN]);
+			xfer.outsig = 1;
+			g_xdag_state = XDAG_STATE_XFER;
+			g_xdag_xfer_last = time(0);
+			xdag_traverse_our_blocks(&xfer, &xfer_callback);
+			
+			cJSON * ret = NULL;
+			cJSON* item = cJSON_CreateObject();				
+			cJSON_AddItemToObject(item, "block", cJSON_CreateString(xdag_hash2address(xfer.transactionBlockHash)));
+			
+			ret = cJSON_CreateArray();
+			cJSON_AddItemToArray(ret, item);
+			return ret;
+		}
+		
+		return NULL;
+	}
+}
+
+#define rpc_register_func(command) xdag_rpc_service_register_procedure(&method_##command, #command, NULL);
+
+/* init rpc procedures */
+int xdag_rpc_init_procedures(void)
+{
+	/* register query func */
+	rpc_register_func(account);
```

Comment:
How these functions work?

Reply:
@jonano614 

Removed useless procedures, if we need it we can add them later.
Only keep five procedures.
They are described in PR description.